cmd/vat/manager/tableDisplay.go

```diff
@@ -0,0 +1,92 @@
+package manager
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/display"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/elrond-go/cmd/vat/evaluation"
+)
+
+var log = logger.GetOrCreate("vat")
+
+type TableFormatter struct {
+	header    []string
+	dataLines []*display.LineData
+}
+
+// to be refactored
+func (tF *TableFormatter) Output(rankedReport RankedReport) error {
+	if rankedReport.NodesAnalyzed == 0 {
+		log.Info("No scanned targets: nothing to display.")
+		return fmt.Errorf("no scanned targets: nothing to display")
+	}
+
+	tF.header = []string{"Index", "Address", "Port", "Status", "Service"}
+
+	for idx, evaluationTarget := range rankedReport.GetAllEvaluatedTargets() {
+		tF.addTargetToTable(idx, evaluationTarget)
+	}
+
+	table, _ := display.CreateTableString(tF.header, tF.dataLines)
+	fmt.Println(table)
+
+	return nil
+}
+
+func (tF *TableFormatter) addTargetToTable(id int, evaluationResult evaluation.EvaluatedTarget) {
+	var line *display.LineData
+
+	if len(evaluationResult.Ports) != 0 {
+		for jdx, tPort := range evaluationResult.Ports {
+			horizontalLineAfter := jdx == len(evaluationResult.Ports)-1
+			if jdx == 0 {
+				line = display.NewLineData(horizontalLineAfter, []string{
+					fmt.Sprintf("%d", id),
+					evaluationResult.Address,
+					fmt.Sprintf("%d", tPort.Number),
+					string(tPort.State),
+					tPort.Protocol})
+			} else {
+				line = display.NewLineData(horizontalLineAfter, []string{
+					"",
+					"",
+					fmt.Sprintf("%d", tPort.Number),
+					string(tPort.State),
+					tPort.Protocol})
+			}
+			tF.dataLines = append(tF.dataLines, line)
+		}
+	} else {
```

Comment:
can remove the else by reverting the if condition and having an early exit

Reply:
thanks. Done.